src/canvas/canvas.service.ts

```diff
@@ -167,6 +167,21 @@ export class CanvasService {
     );
   }
 
+  public async readAllStates(
+    canvasId: Uuid,
+    filter: ListFilter,
+  ): Promise<PagedResult<CanvasStateEntity>> {
+    const qb = PageableUtils.producePagedQueryBuilder(
+      filter,
+      this.canvasStateRepository.createQueryBuilder('state'),
+    );
+    qb.where({ canvasId });
+    return PageableUtils.producePagedResult(
+      filter,
+      await qb.select().getManyAndCount(),
```

Comment:
Is this `select()` needed? I feel like we can safely remove it

```suggestion
      await qb.getManyAndCount(),
```

Reply:
@coredumped7893 You're right. Confirmed, it works.